### DISCUSSION TITLE
Kong Container limiting number of connections

### DISCUSSION THREAD

**Discussion by naveed5** [Kong/kong#7927] [Help] [answered]
Hi All, I have kong container using as load balancer + proxy server and my server application container is running on the same host . I am trying to create 200k websocket connections.

1) I am unable to achieve more than 85k connections when server application and kong containers are on the same host. after reaching 85k I see these kong logs

" *639120 connect() failed (110: Operation timed out) while connecting to upstream, client: 172.16.27.133, server: kong, request: “GET /path HTTP/1.1”, upstream: “http://IP:PORT/”, host: “IP:PORT”

2) I am able to achieve 200k connection when I bypass kong and establish direct connections with server.

3) I am able to achieve 200k connection when I deployed kong container on the other host and my server application on different host .

I have tried changing cache, send, read timeout limits but nothing worked for me . I am curious what’s limiting the number of connections through kong when running on the same host as server application and what are the parameters that needs to be changed.

any help here would be appreciated, please

**Comment by dndx** [comment, marked as answer]
If you run `dmesg` on the container host, is there any `conntrack` errors present? That could be the limiting factor for the number of concurrent connections you can have on a Linux based system.
